### PR TITLE
target: ath79: qca: convert to nvmem-layout 

### PR DESCRIPTION
**Pull request by neheb**
Allows replacing mac-address-increment with mac-base.

ping @Ansuel @rmilecki

### CODE REVIEW

**Comment by rmilecki** [comment]
Did you test it on any of affected devices? In `nvmem-layout` you added `compatible = "nvmem-cells";` which I would **not** expect to work. If it works I think it's by an accident. I believe the correct binding for `nvmem-layout` (that you wanted to use) is `fixed-layout`.

**Comment by neheb** [comment]
Hmm. Could have sworn I copy/pasted that from a different commit. Anyway, I’ll change it.

**Comment by neheb** [comment]
@rmilecki now I'm curious. Would it make sense to convert everything to use nvmem-layout?

**Comment by raenye** [comment]
How would it work with respect to DBDC devices?
I vaguely remember that when the 2nd phy gets created, the increment happens _again_ so they get different MAC addresses.

**Comment by neheb** [comment]
what dbdc devices?

**Comment by raenye** [comment]
> what dbdc devices?

I don't know of any particular `ath79` DBDC devices, but I have a `ramips` device with `mt7615DN` and just wanted to point out the possibility of this PR breaking the behaviour of DBDC devices.

**Comment by neheb** [comment]
ath79 has no dbdc devices.

**Comment by neheb** [comment]
FWIW all nvmem-cells were replaced with nvmem-layout in this commit.

File: target/linux/ath79/dts/qca9531_alfa-network_r36a.dtsi [hunk 1]
```diff
@@ -87,16 +87,21 @@
 				reg = <0x070000 0x010000>;
 				read-only;
 
-				compatible = "nvmem-cells";
-				#address-cells = <1>;
-				#size-cells = <1>;
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9531_alfa-network_r36a.dtsi:90]: It seems that there are some unnecessary new lines. '\n\n\n' can be replaced with '\n\n'.

Comment (neheb) [reply target/linux/ath79/dts/qca9531_alfa-network_r36a.dtsi]: Fixed.

**Comment by neheb** [comment]
I see issues. Draft.

**Comment by rmilecki** [comment]
> Would it make sense to convert everything to use nvmem-layout?

That could be nice but it isn't necessary. If we use `nvmem-cells` without `mac-address-increment` it's a completely valid binding that is compatible with upstream documentation. It will stay supported in kernel for ages.

**Comment by neheb** [comment]
Sure. It does help to be consistent.

**Comment by neheb** [comment]
reworked.

**Comment by rmilecki** [comment]
Looks good to me. Impressive. I vote for merging it. If some single devices fails we can always see into it.

**Comment by neheb** [comment]
will this be merged anytime soon?

**Comment by neheb** [comment]
guess not.

File: target/linux/ath79/dts/qca9558_ubnt_nanobeam-ac-xc.dts [hunk 2]
```diff
@@ -100,11 +100,4 @@
...
 
 &art {
-	compatible = "nvmem-cells";
-	#address-cells = <1>;
-	#size-cells = <1>;
-
-	macaddr_art_0: macaddr@0 {
-		reg = <0x0 0x6>;
-	};
 };
```
Comment (Ansuel) [inline target/linux/ath79/dts/qca9558_ubnt_nanobeam-ac-xc.dts:103]: empty node?

Comment (neheb) [reply target/linux/ath79/dts/qca9558_ubnt_nanobeam-ac-xc.dts:103]: Will fix.

**Pull request merged**